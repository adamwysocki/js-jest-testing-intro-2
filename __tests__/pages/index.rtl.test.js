import { render, fireEvent, cleanup } from "react-testing-library";

import Index from "../../pages";

afterEach(cleanup);

describe("RTL tests", () => {
  it("should render and match snapshot", () => {
    const IndexPage = render(<Index />);
    expect(IndexPage.container).toMatchSnapshot();
  })

  it("should change the language to spanish when the checkbox is checked", () => {
    const IndexPage = render(<Index />);

    const checkbox = IndexPage.getByLabelText("español");

    fireEvent.click(checkbox);

    expect(IndexPage.container).toMatchSnapshot();
  });

  it("should update the welcome message when something is typed", () => {
    const IndexPage = render(<Index />);

    const input = IndexPage.container.querySelector("input#name");
    expect(input).not.toBeNull();

    fireEvent.change(input, {target: { value: "testy timothy" } });

    expect(IndexPage.container).toMatchSnapshot();
  });

  it("should update the welcome message when something is typed and the checkbox is checked", () => {
    const IndexPage = render(<Index />);

    const input = IndexPage.container.querySelector("input#name");
    expect(input).not.toBeNull();

    fireEvent.change(input, {target: { value: "testy timothy" } });

    const checkbox = IndexPage.getByLabelText("español")
    fireEvent.click(checkbox);

    expect(IndexPage.container).toMatchSnapshot();
  });
});