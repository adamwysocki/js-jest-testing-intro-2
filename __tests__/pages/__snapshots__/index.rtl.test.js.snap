// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RTL tests should change the language to spanish when the checkbox is checked 1`] = `
<div>
  <div
    class="container"
    style="margin-top: 150px;"
  >
    <h1>
      ¡Hola!
    </h1>
    <form>
      <div
        class="form-group"
      >
        <input
          aria-describedby="nameHelp"
          class="form-control"
          id="name"
          placeholder="Ingrese su nombre ..."
          type="text"
          value=""
        />
      </div>
      <div
        class="form-check"
      >
        <input
          class="form-check-input"
          id="espanol"
          type="checkbox"
        />
        <label
          class="form-check-label"
          for="espanol"
        >
          español
        </label>
      </div>
    </form>
  </div>
</div>
`;

exports[`RTL tests should render and match snapshot 1`] = `
<div>
  <div
    class="container"
    style="margin-top: 150px;"
  >
    <h1>
      Hello!
    </h1>
    <form>
      <div
        class="form-group"
      >
        <input
          aria-describedby="nameHelp"
          class="form-control"
          id="name"
          placeholder="Enter name ..."
          type="text"
          value=""
        />
      </div>
      <div
        class="form-check"
      >
        <input
          class="form-check-input"
          id="espanol"
          type="checkbox"
        />
        <label
          class="form-check-label"
          for="espanol"
        >
          español
        </label>
      </div>
    </form>
  </div>
</div>
`;

exports[`RTL tests should update the welcome message when something is typed 1`] = `
<div>
  <div
    class="container"
    style="margin-top: 150px;"
  >
    <h1>
      Hello Testy Timothy!
    </h1>
    <form>
      <div
        class="form-group"
      >
        <input
          aria-describedby="nameHelp"
          class="form-control"
          id="name"
          placeholder="Enter name ..."
          type="text"
          value="testy timothy"
        />
      </div>
      <div
        class="form-check"
      >
        <input
          class="form-check-input"
          id="espanol"
          type="checkbox"
        />
        <label
          class="form-check-label"
          for="espanol"
        >
          español
        </label>
      </div>
    </form>
  </div>
</div>
`;

exports[`RTL tests should update the welcome message when something is typed and the checkbox is checked 1`] = `
<div>
  <div
    class="container"
    style="margin-top: 150px;"
  >
    <h1>
      ¡Hola Testy Timothy!
    </h1>
    <form>
      <div
        class="form-group"
      >
        <input
          aria-describedby="nameHelp"
          class="form-control"
          id="name"
          placeholder="Ingrese su nombre ..."
          type="text"
          value="testy timothy"
        />
      </div>
      <div
        class="form-check"
      >
        <input
          class="form-check-input"
          id="espanol"
          type="checkbox"
        />
        <label
          class="form-check-label"
          for="espanol"
        >
          español
        </label>
      </div>
    </form>
  </div>
</div>
`;
